packages/storybook/src/generators/migrate-8/helper-functions.ts

```diff
@@ -0,0 +1,233 @@
+import {
+  applyChangesToString,
+  ChangeType,
+  generateFiles,
+  getPackageManagerCommand,
+  joinPathFragments,
+  output,
+  readProjectConfiguration,
+  Tree,
+  updateProjectConfiguration,
+  workspaceRoot,
+} from '@nx/devkit';
+import { forEachExecutorOptions } from '@nx/devkit/src/generators/executor-options-utils';
+import { tsquery } from '@phenomnomnominal/tsquery';
+import ts = require('typescript');
+import * as fs from 'fs';
+import { fileExists } from 'nx/src/utils/fileutils';
+import { readFileSync } from 'fs';
+import { join } from 'path';
+
+export function onlyShowGuide(storybookProjects: {
+  [key: string]: {
+    configDir: string;
+    uiFramework: string;
+    viteConfigFilePath?: string;
+  };
+}) {
+  const pm = getPackageManagerCommand();
+
+  output.log({
+    title: 'Storybook 8 Migration Guide',
+    bodyLines: [
+      `You can run the following commands manually to upgrade your Storybook projects to Storybook 8:`,
+      ``,
+      `1. Call the Storybook upgrade script:`,
+      `${pm.exec} storybook@latest upgrade`,
+      ``,
+      `2. Call the Storybook automigrate scripts:`,
+      `Run the following commands for each Storybook project:`,
+      ...Object.entries(storybookProjects).map(
+        ([_projectName, storybookProjectInfo]) => {
+          return `${pm.exec} storybook@latest automigrate --config-dir ${storybookProjectInfo.configDir}`;
+        }
+      ),
+      ``,
+    ],
+  });
+}
+
+export function normalizeViteConfigFilePathWithTree(
+  tree: Tree,
+  projectRoot: string,
+  configFile?: string
+): string {
+  return configFile && tree.exists(configFile)
+    ? configFile
+    : tree.exists(joinPathFragments(`${projectRoot}/vite.config.ts`))
+    ? joinPathFragments(`${projectRoot}/vite.config.ts`)
+    : tree.exists(joinPathFragments(`${projectRoot}/vite.config.js`))
+    ? joinPathFragments(`${projectRoot}/vite.config.js`)
+    : undefined;
+}
+
+export function getAllStorybookInfo(tree: Tree): {
```

Comment:
Didn't need the util in the end. It would have added additional complexity and an approach that wasn't necessary.

Instead, followed the same approach the `@nx/storybook/plugin` plugin uses to identify storybook projects.